File: app/models/candidate.rb
class Candidate < ApplicationRecord
  validates :name, presence: true
  validates :email, presence: true, uniqueness: true, format: { with: URI::MailTo::EMAIL_REGEXP }
  validates :date_of_birth, presence: true
  validate :date_of_birth_in_past

  private

  def date_of_birth_in_past
    errors.add(:date_of_birth, 'must be in the past') if date_of_birth && date_of_birth > Date.today
  end
end

File: app/javascript/src/models/candidate.js
import dayjs from 'dayjs'

export default (candidate) => ({
  id: candidate.id,
  name: candidate.name,
  email: candidate.email,
  birthdate: dayjs(candidate.birthdate),

  toForm: () => ({
    name: candidate.name,
    email: candidate.email,
    birthdate: dayjs(candidate.birthdate),
    errors: {},
  }),
})

export const candidateMeta = (meta) => ({
  page: meta.page,
  perPage: meta.per_page,
  totalCount: meta.total_count,
  totalPages: meta.total_pages,
})
